### PR TITLE
Support partial-URI (relative URL) in Web Bundle format

### LINKED ISSUE #14: Needs Clarification: A relative URL is supposed to work in index section?
The context is https://github.com/WICG/webpackage/issues/647

The current format says:

https://wpack-wg.github.io/bundled-responses/draft-ietf-wpack-bundled-responses.html#section-2.2-1

>  Representations within a bundle are named by their Content-Location (Section 8.7 of [I-D.ietf-httpbis-semantics]), which holds a URL. This is also known as the representation's URL.

https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-semantics-18#section-8.7
>     Content-Location = absolute-URI / partial-URI
 
Given that, it seems the format already allows a *relative URL* in the index section, however, it'd be better to ask a clarification than guessing its intention.

Using a relative URL is supposed to work? If so, we don't need to request a bundle format change to resolve https://github.com/WICG/webpackage/issues/647.

### PR DESCRIPTION
**Pull request by horo-t**
Closes https://github.com/wpack-wg/bundled-responses/issues/14

### CODE REVIEW

File: draft-ietf-wpack-bundled-responses.md [hunk 1]
```diff
@@ -85,8 +85,9 @@ Bundle parsers support two primary operations:
 ## Naming a representation {#semantics-naming}
 
 Representations within a bundle are named by their `Content-Location` (Section
-8.7 of {{!I-D.ietf-httpbis-semantics}}), which holds a URL. This is also known
-as the representation's URL.
+8.7 of {{!I-D.ietf-httpbis-semantics}}). This is also known as the
+representation's URL. It is either an absolute-URI or a partial-URI. In the
+latter case, the referenced URL is relative to the bundle's URL.
```
Comment (jyasskin) [inline draft-ietf-wpack-bundled-responses.md]:
```suggestion
latter case, the referenced URL is relative to the bundle's URL unless a section in the bundle specifies a different base URL.
```

To allow a separate spec to override this. e.g. the [primary-url section](https://github.com/WICG/webpackage/blob/main/extensions/primary-section.md) should do so.

File: draft-ietf-wpack-bundled-responses.md [hunk 2]
```diff
@@ -85,8 +85,9 @@ Bundle parsers support two primary operations:
 ## Naming a representation {#semantics-naming}
 
 Representations within a bundle are named by their `Content-Location` (Section
-8.7 of {{!I-D.ietf-httpbis-semantics}}), which holds a URL. This is also known
-as the representation's URL.
+8.7 of {{!I-D.ietf-httpbis-semantics}}). This is also known as the
+representation's URL. It is either an absolute-URI or a partial-URI. In the
```
Comment (jyasskin) [inline draft-ietf-wpack-bundled-responses.md]: I'm not certain whether to cite `absolute-URI or a partial-URI ({{Section 2.7 of RFC7230}})` or `valid url string without a fragment ({{URL}})` here, but do include the link to the specification that defines the term.

Comment (horo-t) [reply draft-ietf-wpack-bundled-responses.md]: Done

File: draft-ietf-wpack-bundled-responses.md (hunk 1 above)
Comment (horo-t) [reply draft-ietf-wpack-bundled-responses.md]: Sounds good.
Done

**Review by horo-t — COMMENTED** [review]
Thank you very much for reviewing.

**Review by jyasskin — APPROVED** [review]
Looks good, but please do run this by wpack@ietf.org before merging.

**Comment by horo-t** [comment]
Thanks. Sent https://mailarchive.ietf.org/arch/msg/wpack/myHwuSoKVdqRxHC6YKCSvPvfpTs/

File: draft-ietf-wpack-bundled-responses.md [hunk 3]
```diff
@@ -85,8 +85,10 @@ Bundle parsers support two primary operations:
 ## Naming a representation {#semantics-naming}
 
 Representations within a bundle are named by their `Content-Location` (Section
-8.7 of {{!I-D.ietf-httpbis-semantics}}), which holds a URL. This is also known
-as the representation's URL.
+8.7 of {{!I-D.ietf-httpbis-semantics}}). This is also known as the
+representation's URL. It is either an absolute-URI or a partial-URI (Section
+2.7 of {{!RFC7230}}). In the latter case, the referenced URL is relative to the
+bundle's URL unless a section in the bundle specifies a different base URL.
```
Comment (felipeerias) [inline draft-ietf-wpack-bundled-responses.md]: This line would be clearer if it specified that it is not talking about any of the default sections defined in the spec, but about an additional one defined by an extension.

I am not sure about the best way to express this idea, so take this as a suggestion:

```suggestion
bundle's URL unless an additional section in the bundle specifies a different
base URL.
```

**Review by felipeerias — COMMENTED** [review]
Thank you, this change looks good to me. I have a small suggestion below.

File: draft-ietf-wpack-bundled-responses.md (hunk 3 above)
Comment (horo-t) [reply draft-ietf-wpack-bundled-responses.md]: Done.

**Review by horo-t — COMMENTED** [review]
Thanks!

**Pull request merged**